### PR TITLE
Link to example file

### PR DESCRIPTION
**Pull request by isoos**
#2957

### CODE REVIEW

File: app/lib/frontend/templates/package.dart [hunk 1]
```diff
@@ -309,9 +309,12 @@ List<Tab> _pkgTabs(
     final exampleFilename = selectedVersion.example.filename;
     renderedExample = renderFile(selectedVersion.example, baseUrl);
     if (renderedExample != null) {
-      renderedExample = '<p style="font-family: monospace">'
-          '<b>${htmlEscape.convert(exampleFilename)}</b>'
-          '</p>\n'
+      final url = getRepositoryUrl(baseUrl, exampleFilename);
```
Comment (sigurdm) [inline app/lib/frontend/templates/package.dart]: Handle getRepositoryUrl returning null?

Comment (sigurdm) [reply app/lib/frontend/templates/package.dart]: Sorry you do that below

File: app/lib/package/models.dart [hunk 2]
```diff
@@ -550,4 +550,15 @@ class PackageLinks {
       issueTrackerUrl: issueTrackerUrl,
     );
   }
+
+  /// URL to use as for linking the referenced resource relative to.
```
Comment (sigurdm) [inline app/lib/package/models.dart]: Broken sentence

File: app/lib/frontend/templates/package.dart [hunk 3]
```diff
@@ -309,9 +309,12 @@ List<Tab> _pkgTabs(
...
+      final escapedName = htmlEscape.convert(exampleFilename);
+      final link = url == null
+          ? escapedName
+          : '<a href="$url" target="_blank" rel="noopener noreferrer nofollow">$escapedName</a>';
+      renderedExample = '<p style="font-family: monospace"><b>$link</b></p>\n'
```
Comment (sigurdm) [inline app/lib/frontend/templates/package.dart]: Side question: why don't we style the link with css?

File: app/lib/package/models.dart [hunk 4]
```diff
@@ -550,4 +550,15 @@ class PackageLinks {
...
+  String get baseUrl {
+    String url = repositoryUrl ?? homepageUrl;
+    if (repositoryUrl != null &&
+        homepageUrl != null &&
+        homepageUrl.startsWith(repositoryUrl)) {
+      url = homepageUrl;
```
Comment (jonasfj) [inline app/lib/package/models.dart]: Hmm... I'm not sure this is a good idea.

We should always use the `repositoryUrl` to identify the repository, **only** if there is no `repositoryUrl`, then we should try to derive one from the the `homepage`.

Adding a heuristic that tries to derive sub-folders from the `homepage` is just more confiusing. If we want that, we should define how to point to sub-folders in the `repositoryUrl` (I'm not sure this is easy to do, we would need to update docs for `pubspec.yaml`).

I think better linking to repository source is a project we could solve, but it should be a project that involves known git hash of the version too.

Comment (jonasfj) [reply app/lib/package/models.dart]: IMO; if we want to do a heuristtic for this, we should do it the same way we do for `LICENSE`. Though ideally, we should implement code listing.

Comment (isoos) [reply app/lib/package/models.dart]: I'll just remove the commit in question from this PR, and address that part later.

File: app/lib/frontend/templates/package.dart (hunk 3 above)
Comment (isoos) [reply app/lib/frontend/templates/package.dart]: It is a good PR to do :)

**Pull request merged**